Gsoc 2025 formatting (#4015)
kubeflow/website: content/en/events/upcoming-events/gsoc-2025.md

__Difficulty:__ Medium

__Size:__ 175 hours

__Goals:__

- Secure our service mesh with istio-cni by default ([`kubeflow/manifests#2907`](https://github.com/kubeflow/manifests/pull/2907))
- Provide an out-of-box option for istio-ambient mesh ([`kubeflow/manifests#2676`](https://github.com/kubeflow/manifests/issues/2676))
   - Controllers to create HTTPRoute and AuthorizationPolicies, that align with way-point proxies
   - Manifests to also have a flavour of HTTPRoute and updated AuthorizationPolicies
- Secure Kserve by default ([`kubeflow/manifests#2811`](https://github.com/kubeflow/manifests/issues/2811))
- Rootless Kubeflow ([`kubeflow/manifests#2528`](https://github.com/kubeflow/manifests/issues/2528))

__Skills Required/Preferred:__

- GitHub and GitHub Actions
- Kubernetes and networking
- Istio, Kustomize

---
---

### Project 4: Deploying Kubeflow with Helm

__Components:__ Kubeflow Manifests, Kubeflow Pipelines, Kubeflow Trainer, Kubeflow Katib, Kubeflow Spark Operator, Kubeflow Model Registry

__Possible Mentors:__ [`@chasecadet`](https://github.com/chasecadet), [`@varodrig`](https://github.com/varodrig), [`@juliusvonkohout`](https://github.com/juliusvonkohout)

> __Difficulty:__ Medium

Reviewer: Lack of newline breaks formatting:

```suggestion

__Difficulty:__ Medium
```

__Skills Required/Preferred:__

- Proficiency in Python, Java and familiarity with developing SDKs.
- Experience with Kubernetes and managing containerized applications.
- Understanding of Apache Spark and its deployment on Kubernetes clusters.
- Familiarity with RESTful API development and integration.
- Experience with monitoring tools and logging frameworks is a plus.

---
---

### Project 7: GPU Testing for LLM Blueprints

__Components:__ Kubeflow Trainer (Training Operator)

__Possible Mentors:__ [`@andreyvelich`](https://github.com/andreyvelich), [`@varodrig`](https://github.com/varodrig)

__Difficulty:__ Medium

__Size:__ 350 hours

__Goals:__

- Explore using Self-Hosted Runners for GPU testing in Kubeflow Trainer ([`kubeflow/trainer#2432`](https://github.com/kubeflow/trainer/issues/2432))

__Skills Required/Preferred:__

- GitHub Actions
- Kubernetes
- PyTorch
- Python

---
---

### Project 8: Support JAX and TensorFlow Training Runtimes

__Components:__ Kubeflow Trainer (Training Operator)

__Possible Mentors:__ [`@andreyvelich`](https://github.com/andreyvelich)

__Difficulty:__ Hard

__Size:__ 350 hours

__Goals:__

> - Add support for JAX and TensorFlow training runtimes in Kubeflow Trainer (`TBA`)

Reviewer: Based on https://github.com/kubeflow/website/pull/4015/files#r1966789839:

```suggestion
- Add support TensorFlow as a training runtime in Kubeflow Trainer ([`kubeflow/trainer#2443`](https://github.com/kubeflow/trainer/issues/2443))
- Add support JAX as a training runtime in Kubeflow Trainer ([`kubeflow/trainer#2442`](https://github.com/kubeflow/trainer/issues/2442))
```

#### Skills required: 
- GitHub Actions 
- Kubernetes 
- PyTorch 
- Explore using Self-Hosted Runners for GPU testing in Kubeflow Trainer ([`kubeflow/trainer#2432`](https://github.com/kubeflow/trainer/issues/2432))

__Skills Required/Preferred:__

- GitHub Actions
- Kubernetes
- PyTorch
- Python

#### Difficulty: hard
---
---

#### Mentors: 
- Andrey Velichkevich, 
- Valentina Rodriguez Sosa
### Project 8: Support JAX and TensorFlow Training Runtimes

#### Length: 350 hrs
__Components:__ Kubeflow Trainer (Training Operator)

---
__Possible Mentors:__ [`@andreyvelich`](https://github.com/andreyvelich)

__Difficulty:__ Hard

__Size:__ 350 hours

__Goals:__

### Project 8
### Title: Support JAX and/or TensorFlow Training Runtimes in Kubeflow Trainer
#### Component: Kubeflow Trainer
- Add support for JAX and TensorFlow training runtimes in Kubeflow Trainer (`TBA`)

> Tracking issue: TBD

Author: Please make a suggestion that I can commit

> Tracking issue: TBD

Reviewer: @juliusvonkohout I made one here: https://github.com/kubeflow/website/pull/4015/files#r1966850541

---

### Project 10: Export Kubeflow Trainer Models to Kubeflow Model Registry

__Components:__ Kubeflow Trainer, Kubeflow Model Registry

> __Possible Mentors:__ [`@Electronic-Waste`](https://github.com/Electronic-Waste)

Author: Please make a suggestion that I can commit

> __Possible Mentors:__ [`@Electronic-Waste`](https://github.com/Electronic-Waste)

Reviewer: Here is one:

```suggestion
__Possible Mentors:__ [`@tarilabs`](https://github.com/tarilabs), [`@franciscojavierarceo`](https://github.com/franciscojavierarceo)
```

__Difficulty:__ Hard

__Size:__ 350 hours

__Goals:__

> - Integrate Kubeflow Trainer with Kubeflow Model Registry (`TBA`)

Reviewer: Based on https://github.com/kubeflow/website/pull/4015/files#r1966791218:

```suggestion
- Integrate Kubeflow Trainer with Kubeflow Model Registry ([`kubeflow/trainer#2438`](https://github.com/kubeflow/trainer/issues/2438))
```

   - Trainer has implemented initializers for model and dataset, and will support model exporter in the future. 

>    - By supporting the model registry as one of the destinations of the exporter, Trainer will integrate with Kubeflow ecosystem more deeply.

Author: Please make a suggestion that I can commit

>    - By supporting the model registry as one of the destinations of the exporter, Trainer will integrate with Kubeflow ecosystem more deeply.

Reviewer: I made one here: https://github.com/kubeflow/website/pull/4015/files#r1966851118

__Skills Required/Preferred:__

- Kubernetes 
- Go 
- YAML
- Python

---
---

### Project 11: Support Volcano Scheduler in Kubeflow Trainer

__Components:__ Kubeflow Trainer

__Possible Mentors:__ [`@Electronic-Waste`](https://github.com/Electronic-Waste)

__Difficulty:__ Hard

__Size:__ 350 hours

__Goals:__

> - Integrate Volcano Scheduler with Kubeflow Trainer (`TBA`)

Reviewer: Based on https://github.com/kubeflow/website/pull/4015/files#r1966791031:

```suggestion
- Integrate Volcano Scheduler with Kubeflow Trainer ([`kubeflow/trainer#2437`](https://github.com/kubeflow/trainer/issues/2437))
```

  - Currently, Trainer does not support Volcano for scheduling. 

>   - Since Volcano is a widely adopted scheduler for AI workloads, it could provide Trainer with more AI-specific scheduling capabilities if we integrate Volcano into Trainer

Author: Please make a suz that I can commit

>   - Since Volcano is a widely adopted scheduler for AI workloads, it could provide Trainer with more AI-specific scheduling capabilities if we integrate Volcano into Trainer

Reviewer: I made one here :https://github.com/kubeflow/website/pull/4015#discussion_r1966851917
